asyncio_run_in_process/main.py

```diff
@@ -193,7 +201,12 @@ async def _monitor_state(
     logger.debug("Waiting for returncode from %s", proc)
     await proc.wait_returncode()
 
-    if proc.returncode == 0:
+    if isinstance(result, InvalidDataFromChild):
+        # When we're unable to unpickle the result from the child, we need to force an error
+        # so that the InvalidDataFromChild is raised in .wait_result_or_raise().
+        proc.error = result
+        proc.returncode = -99
```

Comment:
Any reason not to just use something like `1` here?  I've never found any real "guide" on how to properly use return codes....

Reply:
We use 1 to signal an unexpected exception from the child, and I wanted to use something other than that because in theory the child's result may be anything, and not just an unpickleable custom exception.